dlib/cmake_utils/dlibConfig.cmake.in

```diff
@@ -1,53 +1,51 @@
-# ===================================================================================
-#  The dlib CMake configuration file
-#
-#             ** File generated automatically, do not modify **
-#
-#  Usage from an external project:
-#    In your CMakeLists.txt, add these lines:
-#
-#    find_package(dlib REQUIRED)
-#    target_link_libraries(MY_TARGET_NAME dlib::dlib)
-#
-# ===================================================================================
-
-
-
- 
-# Our library dependencies (contains definitions for IMPORTED targets)
-if(NOT TARGET dlib-shared AND NOT dlib_BINARY_DIR)
-   # Compute paths
-   get_filename_component(dlib_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
-   include("${dlib_CMAKE_DIR}/dlib.cmake")
-   # Check if Threads::Threads target is required and find it if necessary
-   get_target_property(dlib_deps_threads_check dlib::dlib INTERFACE_LINK_LIBRARIES)
-   list(FIND dlib_deps_threads_check "Threads::Threads" dlib_deps_threads_idx)
-   if (${dlib_deps_threads_idx} GREATER -1)
-      if (NOT TARGET Threads)
-         find_package(Threads REQUIRED)
-      endif()
-   endif()
-   unset(dlib_deps_threads_idx)
-   unset(dlib_deps_threads_check)
-endif()
-
-set(dlib_LIBRARIES dlib::dlib)
-set(dlib_LIBS      dlib::dlib)
-set(dlib_INCLUDE_DIRS "@CMAKE_INSTALL_FULL_INCLUDEDIR@" "@dlib_needed_includes@")
-
-mark_as_advanced(dlib_LIBRARIES)
-mark_as_advanced(dlib_LIBS)
-mark_as_advanced(dlib_INCLUDE_DIRS)
-
-# Mark these variables above as deprecated.
-function(__deprecated_var var access)
-   if(access STREQUAL "READ_ACCESS")
-      message(WARNING "The variable '${var}' is deprecated!  Instead, simply use target_link_libraries(your_app dlib::dlib).  See http://dlib.net/examples/CMakeLists.txt.html for an example.")
-   endif()
-endfunction()
-variable_watch(dlib_LIBRARIES __deprecated_var)
-variable_watch(dlib_LIBS __deprecated_var)
-variable_watch(dlib_INCLUDE_DIRS __deprecated_var)
+@PACKAGE_INIT@
 
+include(CMakeFindDependencyMacro)
```

Comment:
Yeah it's screwy that these deps are repeated but it's simple so it's not the end of the world.

Reply:
I'm just so dissatisfied with this. You would think cmake understands enough about a target that by the time you've built it it understands how to install it with one line of code.
Libraries like https://github.com/boost-cmake/bcm.git and https://github.com/robotology/ycm sort of provide this if you use their custom functions.

Reply:
Yeah, it is annoying.  It's also a bit of a regression compared to how it was before.  But it's not a big deal at the end of the day, especially compared to how much other stuff is simpler so ¯\_(ツ)_/¯

Reply:
Though it doesn't fix `dlib.pc` for pkg-config. That needs specific compile and linker options. It there was a magic cmake macro/function which took all your imported targets, generator expressions, and resolved everything down to concrete compile and linker options, then everything would be good. It must do this anyway since it has to generate make/ninja files. If there was a way to reuse those functions then it would be great.

Reply:
Ah, yeah you haven't figured out how to make it generate a `dlib.pc` file yet?

Reply:
No sorry. I mean it generates it fine but it references libraries like blah::blah

Reply:
I think I'll have to use something like file(GENERATE) as a hack

Reply:
Doesn't work.